File: src/main/java/org/pyjjs/scheduler/core/api/impl/actors/resource/supervisor/ResourceSupervisor.java
package org.pyjjs.scheduler.core.api.impl.actors.resource.supervisor;

import org.pyjjs.scheduler.core.api.impl.actors.common.behaviours.BehaviourBasedActor;
import org.pyjjs.scheduler.core.api.impl.actors.resource.supervisor.behaviours.CreateResourceBehaviour;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class ResourceSupervisor extends BehaviourBasedActor<ResourceSupervisorState> {

    private static final Logger LOG = LoggerFactory.getLogger(ResourceSupervisor.class);

    @Override
    protected ResourceSupervisorState getInitialState() {
        ResourceSupervisorState resourceSupervisorState = new ResourceSupervisorState(getContext());
        resourceSupervisorState.setInitialized(true);
        return resourceSupervisorState;
    }

    @Override
    protected void init() {
        fillBehaviours();
        LOG.info("Resource Supervisor initialized");
    }

    private void fillBehaviours() {
        addBehaviour(CreateResourceBehaviour.class);
    }
}

File: src/main/kotlin/org/pyjjs/scheduler/core/api/impl/actors/resource/supervisor/ResourceSupervisor.kt
package org.pyjjs.scheduler.core.api.impl.actors.resource.supervisor

import org.pyjjs.scheduler.core.api.impl.actors.common.behaviours.BehaviourBasedActor
import org.pyjjs.scheduler.core.api.impl.actors.resource.supervisor.behaviours.CreateResourceBehaviour

class ResourceSupervisor : BehaviourBasedActor<ResourceSupervisorState>() {

    init {
        val resourceSupervisorState = ResourceSupervisorState(context)
        updateActorState(resourceSupervisorState)
    }

    override fun init() {
        fillBehaviours()
        LOG.info("Resource Supervisor initialized")
    }

    private fun fillBehaviours() {
        addBehaviour(CreateResourceBehaviour::class)
    }
}
